### PR TITLE
Add Travis jobs for nightly Rust and OpenJDK EA [ECR-3462]

### PR DESCRIPTION
**Pull request by vitvakatu**
## Overview

---
See: https://jira.bf.local/browse/ECR-XYZ

### Definition of Done

- [ ] There are no TODOs left in the code
- [ ] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [ ] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [ ] Public API has Javadoc
- [ ] Method preconditions are checked and documented in the Javadoc of the method
- [ ] Changelog is updated if needed (in case of notable or breaking changes)
- [ ] The continuous integration build passes

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -55,6 +55,20 @@ jobs:
       os: linux
       jdk: openjdk13
       env: CHECK_RUST=false
+    - name: "Linux JDK EA CHECK_RUST=false"
+      os: linux
+      jdk: openjdk-ea
+      env: CHECK_RUST=false
+      before_script:
+        - if [[ "$TRAVIS_EVENT_TYPE" != "cron" ]]; then travis_terminate 0; fi
```
Comment (dmitry-timofeev) [inline .travis.yml]: Interesting, I remember the warning of "don't use exit 1", is this the right thing to use? If so, we shall update our "skip if markdown changes" script.

Comment (slowli) [inline .travis.yml]: Can this job filtering be accomplished with the [conditional jobs](https://docs.travis-ci.com/user/conditional-builds-stages-jobs) (namely, `type = cron` [condition](https://docs.travis-ci.com/user/conditions-v1))? Would be more idiomatic, IMO.

Comment (dmitry-timofeev) [reply .travis.yml]: That would be brilliant, because just firing the VM and performing before_install takes about 2 min (x 3).

**Comment by dmitry-timofeev** [comment]
@vitvakatu Can we test the three new jobs _do_ work?

**Comment by dmitry-timofeev** [comment]
@vitvakatu Also, how the notification will work (email? Mattermost?)?

File: .travis.yml [hunk 2]
```diff
@@ -111,6 +109,7 @@ before_cache:
   - rm -rf "$TRAVIS_HOME/.cargo/registry/src"
 
 before_install:
+  - if [[ -n "$ONLY_CRON_JOB" && "$TRAVIS_EVENT_TYPE" == "cron" ]]; then travis_terminate 0; fi
```
Comment (dmitry-timofeev) [inline .travis.yml]: Nit: CRON_ONLY_JOB sounds more natural (e.g., https://dictionary.cambridge.org/dictionary/english/internet-only , cash-only, read-only, etc.)

File: .travis.yml [hunk 3]
```diff
@@ -136,3 +135,12 @@ install: true  # Skip the installation step, as Maven requires
 script:
   - cd "$TRAVIS_BUILD_DIR"
   - .travis/run_travis_job.sh
+
+notifications:
+  email:
+    - if: type=cron
+    - on_success: always
```
Comment (dmitry-timofeev) [inline .travis.yml]: Will the default (on_failure I suppose) email notification remain for branch builds?

File: .travis.yml [hunk 4]
```diff
@@ -113,3 +136,8 @@ install: true  # Skip the installation step, as Maven requires
 script:
   - cd "$TRAVIS_BUILD_DIR"
   - .travis/run_travis_job.sh
+
+notifications:
+  email:
```
Comment (vitvakatu) [inline .travis.yml]: Email notifications are not triggered for Pull Requests builds, so we will (hopefully) receive notifications for cron builds and master builds only.

**Pull request merged**